src/include/txn.i

```diff
@@ -854,22 +872,16 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint
     if (stop.txnid != WT_TXN_MAX && stop.timestamp != WT_TS_MAX &&
       (!WT_IS_HS(S2BT(session)) || !F_ISSET(session, WT_SESSION_IGNORE_HS_TOMBSTONE)) &&
       __wt_txn_visible(session, stop.txnid, stop.timestamp)) {
-        __wt_buf_free(session, &buf);
-        WT_RET(__wt_upd_alloc_tombstone(session, updp, NULL));
-        (*updp)->txnid = stop.txnid;
-        /* FIXME: Reevaluate this as part of PM-1524. */
-        (*updp)->durable_ts = (*updp)->start_ts = stop.timestamp;
-        F_SET(*updp, WT_UPDATE_RESTORED_FROM_DISK);
+        cbt->upd_value->buf.data = NULL;
+        cbt->upd_value->buf.size = 0;
+        cbt->upd_value->start_ts = stop.timestamp;
+        cbt->upd_value->txnid = stop.txnid;
+        cbt->upd_value->type = WT_UPDATE_TOMBSTONE;
+        cbt->upd_value->prepare_state = WT_PREPARE_INIT;
```

Comment:
@kommiharibabu 
I just had a question about this bit (no need to review the full PR).
There is an existing `FIXME` here to review this as part of the prepare project. I think after this refactor, we should check the prepare flag to know whether this is `WT_PREPARE_INIT` or `WT_PREPARE_RESOLVED`.

I don't think there's any need to block this PR on this since there was an existing `FIXME` for this. But I wanted to flag this to you to figure out whether I'm right and should make a ticket for this.